src/wasm-interpreter.h

```diff
@@ -4953,11 +4978,14 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
           assert(finder.type.isConcrete());
           assert(finder.type.size() >= 1);
           // The continuation is the final value/type there.
-          auto cont = self()->getCurrContinuation();
-          cont->type = finder.type[finder.type.size() - 1].getHeapType();
+          auto newCont = self()->getCurrContinuation();
+          newCont->type = finder.type[finder.type.size() - 1].getHeapType();
+          // And we can set the function to be called, to resume it from here
+          // (the same function we called, that led to a suspension).
```

Comment:
That doesn't seem right. If I resume and then suspend, I may or may not end up in the same `resume` I started at. I could end up at a different `resume` further up or down the stack.

Reply:
I think we are saying the same thing. When a resume is handled, that is the place we can later continue it from. And yes, that may be further up the stack than the original resume that executed it.

Reply:
But where did `contData->func` come from?

Reply:
It is the code that we execute from here:
```
(resume (..handling..)
  (local.get $continuation)
)
```
The `contData->func` comes from that continuation that we are executing. If we handle it here, then to resume it, we'll once more start execution from this exact point. So the same func is exactly what we want.

Reply:
After an offline chat, I understand that this is taking advantage of the property that the originally resumed continuation and the suspending continuation we are handling must have had the same entry point. My confusion partly came from the fact that this property does not hold in the presence of `switch`, but we'll cross that bridge if/when we get to it.